trieste/data.py

```diff
@@ -108,3 +110,65 @@ def astuple(self) -> tuple[TensorType, TensorType]:
         :return: A 2-tuple of the :attr:`query_points` and :attr:`observations`.
         """
         return self.query_points, self.observations
+
+
```

Comment:
After discussion decided against this

Reply:
Can you clarify the choice here?

It seems to me as well that something like a `FidelityDataset` would facilitate things rather than having all these functions to manipulate data, why not maintaining `Mapping[int, Dataset]` for the fidelity levels? would it make all that compute in `predict` and `sample` more expensive? 
or perhaps an object with additional tensor that records fidelity, adding it to the query points does not seem to me it has any benefits?

Reply:
Yeah, I think this is the better way to do it, but I was told not to do this when I was initially implementing it. It was mentioned that this made the typing quite tricky, and might require changes throughout the repo, for relatively little actual gain.

 I'm happy to change it, but it's a fairly large change I think.

Reply:
with "this" you mean current implementation that you have is better?

@uri-granta I guess you had reasons against the above route due to typing?

Reply:
No sorry, I mean the way that you suggested feels much neater